server/build/webpack.js

```diff
@@ -206,6 +218,10 @@ export default async function createCompiler (dir, { buildId, dev = false, quiet
     loader: 'emit-file-loader',
     include: [dir, nextPagesDir],
     exclude (str) {
+      if (shouldTranspileModule(str)) {
```

Comment:
This works for sure.
For the SSR version, we may need to rewrite the `require('my-module')` pointing to this emitted file.

And if that requires a ES6+ module, how we are going to handle it?

Reply:
This works for SSR since node will look for `.next/dist/node_modules` first and pick up the transpiled version in it.

> And if that requires a ES6+ module, how we are going to handle it?

During my testing `my-modules` was depending on another es6 module and both were transpiled and emitted to `.next/dist/node_modules/my-modules`. 

Later today or tomorrow (when I have time basically) I will add those to the PR together with the changes to the readme.

As I said this solution still doesn't work with linked packages though - I will try to tackle that at the end and see if it is easy/doable to fix.

Reply:
@giuseppeg 

> This works for SSR since node will look for .next/dist/node_modules first and pick up the transpiled version in it.

That's pretty interesting. Nice work.

Looking forward to take this in.

Reply:
@arunoda *edit* in order for this to work we need to copy the `package.json` over too. Do you have any idea how to do so? Or would you rather rewrite the imports/requires